backend/hitas/models/codes.py

```diff
@@ -0,0 +1,72 @@
+from django.contrib import admin
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+import re
+
+
+def validate_code_number(value):
+  m = re.search(r'^(\d{3})$', value)
+
+  if m is None:
+    raise ValidationError(
+        _('%(value)s is not an valid code number'),
+        params={'value': value},
+    )
```

Comment:
If `validate_code_number` is only to be used inside `Code` it could be its method instead

Reply:
I kept it like this for now. We most likely need it for request processing